bot/exts/help_channels/__init__.py

```diff
@@ -0,0 +1,17 @@
+import logging
+
+from bot.bot import Bot
+
+log = logging.getLogger(__name__)
+
+
+def setup(bot: Bot) -> None:
+    """Load the HelpChannels cog."""
+    # Defer import to reduce side effects from importing the sync package.
+    from bot.exts.help_channels import _cog
```

Comment:
Automatic extension discovery has to import this subpackage to check if it's a valid extension. This is mainly to be able to list all extensions with the command that shows the loaded/unloaded status. Otherwise, it could have taken a more dynamic approach which attempts to load extensions as requested and considers them invalid if discord.py throws an error.

If this extension is not being loaded, then any code that gets executed upon importing the package is just a waste. Side effects could be compiling regex, reading file content, importing expensive third-party packages, etc. I wasn't thinking of anything specific in this case; it was done more as a precaution. In any case, it isn't too effective since the `_channel` module has to be imported to get the constant needed by `validate_config`. Could be avoided if I cared enough, but the gains are not very significant so I'm not sure why I bothered.